ocaml/xapi/xapi_pool.ml

```diff
@@ -1646,12 +1646,14 @@ let sync_database ~__context =
 
 (* This also means me, since call will have been forwarded from the current master *)
 let designate_new_master ~__context ~host =
+  Xapi_pool_helpers.assert_pool_secret_rotation_not_pending ~__context ;
```

Comment:
I think that the allowed operation checking should take care of this instead of having to add N checks for each of the N operation implementations. So these are redundant.
HA is enabled is the exception because it is a permanent state rather than a single operation.

Reply:
See 
```
let with_pool_operation ~__context ~self ~doc ~op f =
  let task_id = Ref.string_of (Context.get_task_id __context) in
  Helpers.retry_with_global_lock ~__context ~doc (fun () ->
      assert_operation_valid ~__context ~self ~op ;
      Db.Pool.add_to_current_operations ~__context ~self ~key:task_id ~value:op) 
```

As long as you taught the assert_operation_valid to reject the incompatible concurrent ops (which you did by declaring the operation and adding it to the list of incompatible ops) that is all you really need to do.

Reply:
I think you're confusing HA enable (the operation) and HA enabled (the state); nonetheless there is some existing code which wants to know if HA enable is in progress, so I'll leave these functions here